src/app/services/actions/query-action-creator-util.ts

```diff
@@ -100,12 +100,16 @@ const makeRequest = (httpVerb: string, scopes: string[]): Function => {
 
     if (query.sampleHeaders && query.sampleHeaders.length > 0) {
       query.sampleHeaders.forEach(header => {
-
-        // We are relying on the graph client to set the content type header.
+        // We are relying on the graph client to set the 'application/json' content type header.
         if (header.name.toLowerCase() === 'content-type') {
-          return;
+          if (header.value !== 'application/json') {
+            sampleHeaders[header.name] = header.value;
+          } else {
+            return;
```

Comment:
the comment above should probably just above this return to indicate why you just return.